pkgs/build-support/setup-hooks/separate-debug-info.sh

```diff
@@ -20,7 +20,7 @@ _separateDebugInfo() {
 
         # Extract the Build ID. FIXME: there's probably a cleaner way.
         local id="$($READELF -n "$i" | sed 's/.*Build ID: \([0-9a-f]*\).*/\1/; t; d')"
-        if [ "${#id}" != 40 ]; then
+        if [ "${#id}" < 3 ]; then
```

Comment:
```suggestion
        if [ "${#id}" < 16 ]; then
```
if llvm produces 16 bytes then we should use that as a lower limit. Otherwise collisions will happen all the time.

Reply:
`<` is a lexicographical compare. You probably want `-gt` for numeric comparison: `if [ "${#id}" -lt 3 ]; then`